engine/src/em-font.cpp

```diff
@@ -0,0 +1,243 @@
+Software Foundation.
+
+LiveCode is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or
+FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+for more details.
+
+You should have received a copy of the GNU General Public License
+along with LiveCode.  If not see <http://www.gnu.org/licenses/>.  */
+
+#include <foundation.h>
+
+#include "prefix.h"
+
+#include "filedefs.h"
+
+#include "system.h"
+
+#include <ft2build.h>
+#include FT_FREETYPE_H
+#include <freetype/ftsnames.h>
+
+#include "customfont.h"
+#include "freetype-font.h"
+#include "skiatypeface.h"
+
+#include "em-font.h"
+
+#define MCEMSCRIPTEN_FONT_FOLDER "/boot/standalone"
+
+bool MCEmscriptenCustomFontsInitialize()
+{
+	return MCCustomFontListInitialize();
+}
+
+void MCEmscriptenCustomFontsFinalize()
+{
+	MCCustomFontListFinalize();
+}
+
+static bool MCEmscriptenCreateCustomFontFromPath(MCStringRef p_path, FT_Library p_library, MCCustomFont *&r_custom_font)
+{
+	bool t_success;
+	t_success = true;
+	
+	MCAutoStringRef t_font_path;
+	
+	MCAutoDataRef t_buffer;
+	if (t_success)
+		t_success = MCS_loadbinaryfile(p_path, &t_buffer);
+	
+	MCCustomFont *t_font;
+	t_font = nil;
+	if (t_success)
+		t_success = MCMemoryNew(t_font);
+	
+	if (t_success)
+		t_success = MCFreeTypeGetFontInfoFromData(p_library, *t_buffer, t_font->name, t_font->family, t_font->style);
+
+	if (t_success)
+	{
+		MCValueAssign(t_font->path, p_path);
+		r_custom_font = t_font;
+	}
+	else
+		MCCustomFontDelete(t_font);
+	
+	return t_success;
+}
+
+static bool MCEmscriptenCustomFontLoadCallback(void *p_context, const MCSystemFolderEntry *p_entry)
+{
+	FT_Library t_ft_library = reinterpret_cast<FT_Library>(p_context);
+
+	if (!p_entry->is_folder)
+	{
+		MCCustomFont *t_font;
+		if (MCEmscriptenCreateCustomFontFromPath(p_entry->name, t_ft_library, t_font))
+			MCCustomFontListAddFont(t_font);
+	}
+
+	return true;
+}
+
+// Parse the fonts folder (as set up by the standalone builder), locating any font files.
+// For each font file found, attempt to parse using freetype and if successful, add to the custom font list.
+// If particularly intensive, the parsing step could be moved to the IDE, which would just generate the required meta-data.
+bool MCEmscriptenCustomFontsLoad()
+{
+	bool t_success;
+	t_success = true;
+	
+	FT_Library t_ft_library;
+	t_ft_library = nil;
+	if (t_success)
+		t_success = FT_Init_FreeType(&t_ft_library) == 0;
+	
+	if (t_success)
+		t_success = MCsystem->ListFolderEntries(MCSTR(MCEMSCRIPTEN_FONT_FOLDER), MCEmscriptenCustomFontLoadCallback, t_ft_library);
+	
+	if (t_ft_library != nil)
+		FT_Done_FreeType(t_ft_library);
+
+	return t_success;
+}
+
+
+bool MCEmscriptenListFontFamilies(MCListRef& r_names)
+{
+	MCAutoListRef t_list;
+	if (!MCListCreateMutable('\n', &t_list))
+		return false;
+	
+	MCAutoStringRef t_custom_font_names;
+	if (!MCCustomFontListGetNames(&t_custom_font_names) ||
```

Comment:
Either do:
```
if (!... ||
        !...)
    return false;
```
or
```
if (... ||
    ...)
{
...   
}
```
i.e. if no braces make sure the condition wraps one level deeper than the body.

Reply:
Fixed